jbmc/src/java_bytecode/java_types.cpp

```diff
@@ -87,6 +87,11 @@ reference_typet java_lang_object_type()
   return java_reference_type(symbol_typet("java::java.lang.Object"));
 }
 
+/// Construct an array pointer type. It is a pointer to a symbol with identifier
+/// `java::array[<primitive type or reference>]` and an element type
+/// which is a primitive type or void* for array of references (this gets
+/// replaced when the runtime type of such array is available).
```

Comment:
This is the static type not the runtime type, and it's worth clarifying the `ID_C_element_type` returned from this path is never more informative than the type name (i.e. `java::array[int]` always has `ID_C_element_type == "int"` come what may, and for this function `array[reference]` always gets `void *`, nothing more informative.

Reply:
I can't really see what information from your comment is missing in the doc... would this be any better? 
Construct an array pointer type from a static type. It is a pointer to a symbol with identifier
 `java::array[<primitive type or reference>]` and an element type which is a primitive type or void* for array of references (element type can be updated when the runtime type of such array becomes available).

Reply:
`(element type can be updated when the runtime type of such array becomes available)` doesn't make sense to me -- are you saying this function might do that, or someone that used this function could potentially do that later?

Reply:
Suggested reformat of the rest:
Construct an array pointer type. It is a pointer to a symbol with identifier
java::array[<primitive type or reference>]. Its `ID_C_element_type` is set to the corresponding primitive type, or void* for arrays of references.

Reply:
typo: `if available` I think? 

The answer to @smowton question is I think it potentially gets set later right?

Reply:
Yes, I meant later on, not in this function. I've seen parts of code which use this function to create array of references and right on the next line they change element type to something specific. This seems to be the only way of creating an array with specific element type. But I agree it might be bringing more confusion than benefit so I'll leave that part out.